sonar-scanner-engine/src/main/java/org/sonar/batch/issue/DefaultFilterableIssue.java

```diff
@@ -61,12 +61,12 @@ public String message() {
 
   @Override
   public Integer line() {
-    return rawIssue.hasLine() ? rawIssue.getLine() : null;
+    return rawIssue.hasTextRange() ? rawIssue.getTextRange().getStartLine() : null;
   }
 
   @Override
   public Double gap() {
-    return rawIssue.hasGap() ? rawIssue.getGap() : null;
+    return rawIssue.getGap() != 0 ? rawIssue.getGap() : null;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Equality tests should not be made with floating point values. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1244)